third_party_integration/huggingface_transformers/0001-Modifications-for-NNCF-usage.patch

```diff
@@ -1301,31 +1301,7 @@ index 214e7a978..530e80048 100755
          # Seed must be set before instantiating the model when using model
          enable_full_determinism(self.args.seed) if self.args.full_determinism else set_seed(self.args.seed)
          self.hp_name = None
-@@ -623,7 +652,10 @@ class Trainer:
-         self.current_flos = 0
-         self.hp_search_backend = None
-         self.use_tune_checkpoints = False
--        default_label_names = find_labels(self.model.__class__)
-+        model_class = self.model.__class__
-+        if isinstance(self.model, NNCFNetwork):
-+            model_class = self.model.get_nncf_wrapped_model().__class__
-+        default_label_names = find_labels(model_class)
-         self.label_names = default_label_names if self.args.label_names is None else self.args.label_names
-         self.control = self.callback_handler.on_init_end(self.args, self.state, self.control)
- 
-@@ -708,7 +740,10 @@ class Trainer:
-     def _set_signature_columns_if_needed(self):
-         if self._signature_columns is None:
-             # Inspect model forward signature to keep only the arguments it accepts.
--            signature = inspect.signature(self.model.forward)
-+            if isinstance(self.model, NNCFNetwork):
-+                signature = inspect.signature(self.model.get_nncf_wrapped_model().forward)
-+            else:
-+                signature = inspect.signature(self.model.forward)
-             self._signature_columns = list(signature.parameters.keys())
-             # Labels may be named label or label_ids, the default data collator handles that.
-             self._signature_columns += list(set(["label", "label_ids"] + self.label_names))
-@@ -1409,6 +1444,8 @@ class Trainer:
+@@ -1409,6 +1438,8 @@ class Trainer:
```

Comment:
Now the integration into transformers won't need special signature and class handling for `NNCFNetwork` objects. 16 lines less in total in 2 places.

Reply:
@vuiseng9 @yujiepan-work, potentially, it may affect NNCF integration to Optimum